src/plugins/sailfishoswizards/desktopeditor.cpp

```diff
@@ -0,0 +1,80 @@
+/****************************************************************************
+**
+** Copyright (C) 2019 Open Mobile Platform LLC
+** Contact: https://community.omprussia.ru/open-source
+**
+** This file is part of Qt Creator.
+**
+** $QT_BEGIN_LICENSE:LGPL$
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 as published by the Free Software
+** Foundation and appearing in the file LICENSE.LGPL3 included
+** in the packaging of this file. Please review the following information
+** to ensure the GNU Lesser General Public License version 2.1 requirements
+** will be met: https://www.gnu.org/licenses/lgpl-2.1.html.
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+#include "desktopeditor.h"
+
+namespace SailfishOSWizards {
+namespace Internal {
+
+using Core::IDocument;
+
+/*!
+ * \brief Constructor for the editor object.
+ * Initializes this editor's \b document and \b widget.
+ * \sa DesktopEditorWidget, DesktopDocument
+ */
+DesktopEditor::DesktopEditor()
+{
+    this->m_editorWidget = new DesktopEditorWidget;
+    this->m_editorDocument = new DesktopDocument(this, this->m_editorWidget);
+    this->setWidget(m_editorWidget);
```

Comment:
Main reason save the point of m_editorWidget thats i can't new DesktopEditorWidget(this), because this  - in not a QWidget. I can dont save the pointer but leave delete widget() in destructor.